calculator/src/main/java/com/market/banica/calculator/service/JMXServiceImpl.java

```diff
@@ -0,0 +1,299 @@
+                , recipeName, ingredientName, newValue);
+    }
+
+    @Override
+    @ManagedOperation
+    public String getProductQuantity(String recipeName, String ingredientName) {
+        LOGGER.debug("JMX server impl: In getProductQuantity method");
+        LOGGER.info("GetProductQuantity called from JMX server for recipe {} and ingredient {}", recipeName, ingredientName);
+
+        Product parentProduct = retrieveProductFromDatabase(recipeName);
+
+        validateProductExist(recipeName, parentProduct);
+
+        if (doesIngredientBelongToRecipe(ingredientName, parentProduct)) {
+
+            LOGGER.info("Value checked from JMX server for recipe {} and ingredient {}", recipeName, ingredientName);
+            return String.valueOf(getProductQuantity(ingredientName, parentProduct));
+
+        } else {
+
+            return throwExceptionWhenProductDoesNotBelongToRecipe(recipeName, ingredientName);
+        }
+    }
+
+    @Override
+    @ManagedOperation
+    public String getUnitOfMeasure(String productName) {
+        LOGGER.debug("JMX server impl: In getUnitOfMeasure method");
+        LOGGER.info("GetUnitOfMeasure called from JMX server for product with name: {}",productName);
+
+        Product product = retrieveProductFromDatabase(productName);
+
+        validateProductExist(productName, product);
+
+        LOGGER.info("UnitOfMeasure checked from JMX server for product with name {}", productName);
+        return product.getUnitOfMeasure().toString();
+    }
+
+    @Override
+    @ManagedOperation
+    public void setUnitOfMeasure(String productName, String unitOfMeasure){
+        LOGGER.debug("JMX server impl: In setUnitOfMeasure method");
+        LOGGER.info("SetUnitOfMeasure called from JMX server for product with name: {} " +
+                "and new UnitOfMeasure: {}",productName, unitOfMeasure);
+
+        Product product = retrieveProductFromDatabase(productName);
+
+        validateProductExist(productName, product);
+
+        validateUnitOfMeasureExist(unitOfMeasure);
+
+        product.setUnitOfMeasure(Enum.valueOf(UnitOfMeasure.class,
+                unitOfMeasure.toUpperCase(Locale.ROOT)));
+
+        productService.createBackUp();
+
+        LOGGER.info("UnitOfMeasure set from JMX server for product with name {}" +
+                " and new unitOfMeasure {}", productName,unitOfMeasure);
+    }
+
+    @Override
+    @ManagedOperation
+    public void deleteProduct(@Nullable String parentProductName, String productName) {
+        LOGGER.debug("JMX server impl: In deleteIngredient method");
+        LOGGER.info("DeleteIngredient called from JMX server for recipe {} and ingredient {}", parentProductName, productName);
+
+        Product product = retrieveProductFromDatabase(productName);
+        Product parentProduct = retrieveProductFromDatabase(productName);
+
+        validateProductExist(productName, product);
+        validateProductExist(productName, product);
```

Comment:
I think you meant to write something else in the second validateProductExist method call

Reply:
Done